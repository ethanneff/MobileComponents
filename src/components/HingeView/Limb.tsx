import React from "react";
import { Line } from "react-native-svg";
import { useBoneMode } from "../../commons/Hooks/useBoneMode";
import { Point2 } from "../../commons/Utils/Maths/geometry/Point";
import Bone from "./Bone";

export interface LimbProps {
  strokeWidth?: number;
  strokeColor: string;
  point1: Point2;
  point2: Point2;
  id?: string;
}

export default function Limb({
  strokeWidth,
  strokeColor,
  point1: vector1,
  point2: vector2,
  id
}: LimbProps) {
  const boneMode = useBoneMode();
  if (boneMode) {
    return (
      <Bone
        strokeColor={strokeColor}
        point1={vector1}
        point2={vector2}
        id={id}
      />
    );
  }
  return (
    <Line
      id={id}
      x1={vector1.x}
      y1={vector1.y}
      x2={vector2.x}
      y2={vector2.y}
      stroke={strokeColor}
      strokeWidth={strokeWidth}
    />
  );
}
