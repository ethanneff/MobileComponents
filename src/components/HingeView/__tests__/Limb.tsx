import React from "react";
import Victor from "victor";
import { testRender } from "../../../commons/Utils/TestMocks/test-render";
import Limb from "../Limb";

describe("Limb Component", () => {
  it("renders correctly", () => {
    const vector1 = new Victor(10, 20);
    const vector2 = new Victor(20, 20);

    const { tree } = testRender(
      <Limb
        id="limb"
        point1={vector1}
        point2={vector2}
        strokeWidth={5}
        strokeColor={"black"}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
