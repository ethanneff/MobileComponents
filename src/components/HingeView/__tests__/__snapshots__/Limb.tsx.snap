// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Limb Component renders correctly 1`] = `
<RNSVGLine
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  name="limb"
  opacity={1}
  originX={0}
  originY={0}
  propList={
    Array [
      "stroke",
      "strokeWidth",
    ]
  }
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={4278190080}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={5}
  vectorEffect={0}
  x={0}
  x1={10}
  x2={20}
  y={0}
  y1={20}
  y2={20}
/>
`;
